### PR TITLE
examples: Switch `docker-compose` -> `docker compose`

### PR DESCRIPTION
**Pull request by phlax**
spot the difference!

- switch to `docker compose` / buildkit
- fix flaky/sleeps 
- assorted cleanups
- speed up verify examples by 20-30%

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by phlax** [comment]
this may cause a few flakes as it works differently to previously - ive fixed a few so hopefully it will be good

this will finally fix the issues that so many people have in #20915 once and for all!

File: examples/vrp-local/verify.sh [hunk 1]
```diff
@@ -1,7 +1,9 @@
 #!/bin/bash -e
 
 export NAME=vrp-local
-export DELAY=10
+export MANUAL=true
+
+export DOCKER_BUILDKIT=0
```
Comment (phlax) [inline examples/vrp-local/verify.sh:6]: add comment

File: .github/dependabot.yml [hunk 2]
```diff
@@ -16,21 +16,11 @@ updates:
   schedule:
     interval: "daily"
 
-- package-ecosystem: "pip"
```
Comment (phlax) [inline .github/dependabot.yml:19]: put these back and move the requirements files into dirs so they are picked up

**Comment by phlax** [comment]
@mattklein123 this will most likely conflict with other examples PRs i have open

This one is defo the priority as it has quite a few benefits, and unblocks other moves.

There are quite a lot of followups/cleanups i would like to make to both docs and examples, but i think this already has quite a lot of change

**Comment by phlax** [comment]
i think this should also fix a few flakes - like this one https://dev.azure.com/cncf/envoy/_build/results?buildId=130154&view=logs&j=f1a49366-51c0-540f-0294-f73c2bb311a9&t=7fd036c9-61ec-5b59-3645-e90950dc67a6&l=16129

**Review by mattklein123 — APPROVED** [review]

**Pull request merged**